src/averageAges.js

```diff
@@ -36,8 +40,18 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const newArray = people.filter(person =>
+    withChildren
+      ? person.sex === 'f' && people.find(child => child.mother === person.name)
```

Comment:
```suggestion
      ? person.sex === 'f' && people.some(child => child.mother === person.name)
```
it is better to use `some`, because we don't need the element itself that the `find` method returns, we just want to know if this element exists